.github/workflows/pytest.yml

```diff
@@ -0,0 +1,41 @@
+name: Python package
+
+on:
+  pull_request:
+    branches:    
+      - main
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.11"]
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: 'pip'
+      # You can test your matrix by printing the current Python version
+      - name: Display Python version
+        run: python -c "import sys; print(sys.version)"
+      - name: Install taskweaver
+        run: |
+          python -m pip install --upgrade pip setuptools wheel
+          pip install -e .
+      - name: Test with pytest
+        run: |
+          pip install pytest pytest-cov
+          pytest tests/unit_tests --doctest-modules --junitxml=junit/test-results.xml --cov=com --cov-report=xml --cov-report=html --ignore tests/unit_tests/data --ignore tests/unit_tests/test_embedding.py --ignore tests/unit_tests/test_plugin_selector.py
```

Comment:
maybe the `--junitxml` parameter need to have python version tag as well

Reply:
For the `ignore` list of test cases, maybe we can mark them in each test case separately so that we do not need to update the yaml when updating test cases? Maybe we can set a command line parameter ( use [`pytest.config.get_option` ](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.Config.getoption) to read it) or env var and check it with a [`@skipif`](https://docs.pytest.org/en/7.1.x/how-to/skipping.html#id1) mark?

Reply:
I have @pytest.mark.skip, not sure why it doesn't work :(